### ISSUE TITLE
ROCm Kernel (Ussing Head with GNOME  X11/GTK) not working with Ubuntu 16.04 (RX480)

### ISSUE THREAD

**Issue by cathalgarvey** [RadeonOpenCompute/ROCK-Kernel-Driver#30]
Hi folks,

I was running ROCm for a few months on this box, but in the last week it's started crashing more and more often, and yesterday it just started freezing shortly after boot-up, particularly if I touched the display settings for my monitors.

Because I wasn't confident the install was clean to begin with, and ROCm had been unstable and unpredictable for compute all that time, I decided to try a fresh install before reporting a bug.

First, I tried avoiding ROCm and just using amdgpu-pro, because I just want an OpenCL 1.2 compatible runtime for now (so, although Mesa supports Polaris and doesn't destroy my system, it won't do as it's 1.1 only :( ). That rendered the system so unstable I couldn't even login, so I wiped and reinstalled again.

So, I have installed Ubuntu GNOME 16.04 fresh, and then followed the ROCm install. But this time I had the presence of mind to make sure I could boot from grub with a stock kernel. When I re-started, the system hung again when I opened the display app, exactly as before.

What information can I collect about this hanging issue to help debug it? It's pretty consistently reproducible this past week, I just boot, login, and open the screen-orientation app to move my two monitors around, and it freezes. It also freezes without reason at any other time, but the display app seems to do it every time. I can boot into the ROCm Kernel, crash out, and get whatever information is needed to help fix this.

**Comment by gstoner** [comment]
We tracked this down to a bug in the base AMDGPU 4.11 Linux kernel driver. I asking the Linux kernel team for fix. Working on a ETA

**Comment by cathalgarvey** [comment]
Great, thanks :)

On 14 September 2017 12:24:31 GMT+01:00, Gregory Stoner <notifications@github.com> wrote:
>We tracked this down to a bug in the base AMDGPU 4.11 Linux kernel
>driver. I asking the Linux kernel team for fix. Working on a ETA
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/RadeonOpenCompute/ROCK-Kernel-Driver/issues/30#issuecomment-329453208

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by cathalgarvey** [comment]
Some more background would help, actually; when the bug would have first been pushed out via apt, for example. Is there a versioning policy which indicates which apt releases are more/less likely to be stable, so that we could pin in apt when things are working fine, and only manually upgrade when required for features or specific bugfixes?

I'm glad to have the apt route to keep up with a fast project like this, but system stability on my daily machine is obviously also a consideration. :)

**Comment by cathalgarvey** [comment]
Hi folks, is there any update here?

**Comment by cathalgarvey** [comment]
Update from my end: I've installed the latest ROCm stack from Apt, but the problem remains. Boot into the ROCm kernel, and within minutes the system freezes unrecoverably.

I'm still using Ubuntu GNOME 16.04.

**Comment by gstoner** [comment]
One thing our standard mode of operation is headless.  But I am will talk to Linux team again about this.

**Comment by cathalgarvey** [comment]
Thanks for the update. In the meantime, I've purged ROCm and I've discovered the wonders of the `--compute` flag for AMDGPU-pro install, which seems to be working without explosions or crashes. :)

**Comment by delbabrour** [comment]
any news on kernel update ?? 
+1
Update to at least linux kernel 4.14 or 4.15 rc1 
Kernel bug here https://bugs.freedesktop.org/show_bug.cgi?id=100443

linux team already provided solution can we have a version built on 4.14

**Comment by MichaelMcCulloch** [comment]
I would like to chime in here.

Adding the kernel parameters "amdgpu.cik_support=1 radeon.cik_support=0"
Has fixed this issue for me. 

You might also try "si_support" instead of "cik_support"

**Comment by gstoner** [comment]
ROCm 1.7 was cut on 4.13 Linux kernel which just coming out, but it is DKMS release so bolt on the standard Linux kernel with the distro. 

Now we have this core development behind us we just move the core development of the driver to Linux kernel 4.15, 

4.15  also we upstream all the KFD changes for Discreet GPU support, they have few more things they working on around memory management for Discreet GPU we need to upstream.  But we close to having key capabilities upstream for ROCm.     https://lists.freedesktop.org/archives/dri-devel/2017-November/156568.html

**Comment by jlgreathouse** [comment]
Circling back to this -- we have moved to DKMS modules as part of ROCm since 1.7, and over the course of 2018 went all the way through a ROCm 2.0 release. Ubuntu 16.04 and Ubuntu 18.04 should both be supported. I just tested ROCm 2.0 on Ubuntu GNOME 16.04.5 LTS, and things appear to be working as expected on my Polaris 10 GPU. This includes features like resetting the GPU, over/underclocking, and running a lot of benchmarks while testing these things. As such, I'm going to close this issue because I suspect any problems are resolved both in our ROCm-provided DKMS driver as well as in the upstream driver.
